app/assets/src/components/Samples.jsx

```diff
@@ -147,11 +147,16 @@ class Samples extends React.Component {
   }
 
   startReportGeneration() {
+    Samples.showLoading('Downloading reports...');
     axios.get(`/projects/${this.state.selectedProjectId}/make_project_reports_csv`).then((res) => {
+      Samples.hideLoader();
```

Comment:
The download is asynchronous, so it is not actually finished when you get the response from this initial request, it just starts the report generation. You should hide the loader only when `status == 'complete'` in the checkReportDownload() function

Reply:
Really help info, thanks @cdebourcy